source/MapDetailPanel.cpp

```diff
@@ -534,6 +533,40 @@ void MapDetailPanel::DrawKey()
 		"Danger level:"
 	};
 	const string &header = HEADER[-min(0, max(-7, commodity))];
+	const vector<Trade::Commodity> &commodities = GameData::Commodities();
+
+	int headerWidth = font.Width(header);
+	int maxWidth = headerWidth;
+	if(commodity >= 0)
+	{
+		const auto &range = commodities[commodity];
+		if(static_cast<unsigned>(commodity) >= commodities.size())
+			return;
+
+		for(int i = 0; i <= 3; ++i)
+		{
+			int price = range.low + ((range.high - range.low) * i) / 3;
+			maxWidth = max(maxWidth, font.Width(to_string(price)));
+		}
+	}
+
+    int longestCommodityWidth = 0;
+    for(const auto &commodity : commodities)
+    {
+        int commodityWidth = font.Width(commodity.name);
+        longestCommodityWidth = max(longestCommodityWidth, commodityWidth);
+    }
+
+    int longestHeaderWidth = 0;
+    for(const auto &header : HEADER)
+    {
+        int headerWidth = font.Width(header);
+        longestHeaderWidth = max(longestHeaderWidth, headerWidth);
+    }
+
+    maxWidth = std::max(maxWidth, std::max(longestCommodityWidth, longestHeaderWidth));
+
+	Point pos = Screen::TopRight() + Point(-20. - maxWidth, 310.);
```

Comment:
What's the purpose of this code change? I understand that we need to change the pos point to account for the new UI, but why use the max width of the headers, commodity names, and commodity prices? Why not just pick a single value that works and update the X value of the pos point where it was already defined?

Notably, the commodity names don't even show up in this UI, so I'm not sure why they're being used to determine the max width. The commodity prices are used, but they're always going to be shorter than "Uninhabited" and "Unexplored", which always show up but aren't being compared against in this code.
![image](https://github.com/endless-sky/endless-sky/assets/17688683/7607f1e4-0bf9-4741-a185-5ba9cac8b048)
There's also a lot more that can show up in the key than just commodity prices and headers that could potentially be longer than what's being measured. The original issue is even an example of that.
![image](https://github.com/endless-sky/endless-sky/assets/17688683/07ce45ac-f4ee-45cf-bbdb-76187f36c557)

Given that the UI image still only has so much space to work with, I think that trying to measure some max width of the text in the key is overengineering it (and if done more accurately, could actually result in someone creating a government name that kicks the entire key outside of the image instead of just chopping off the end of the government name). Just pick an X value that works for the new image.